src/structures/GuildScheduledEvent.js

```diff
@@ -0,0 +1,390 @@
+'use strict';
+
+const Base = require('./Base');
+const {
+  GuildScheduledEventEntityTypes,
+  GuildScheduledEventStatuses,
+  GuildScheduledEventPrivacyLevels,
+  Endpoints,
+} = require('../util/Constants');
+const SnowflakeUtil = require('../util/SnowflakeUtil');
+
+/**
+ * Represents a scheduled event in a {@link Guild}.
+ * @extends {Base}
+ */
+class GuildScheduledEvent extends Base {
+  constructor(client, data) {
+    super(client);
+
+    /**
+     * The id of the guild scheduled event
+     * @type {Snowflake}
+     */
+    this.id = data.id;
+
+    /**
+     * The id of the guild this guild scheduled event belongs to
+     * @type {Snowflake}
+     */
+    this.guildId = data.guild_id;
+
+    /**
+     * Whether the guild scheduled event has been deleted
+     * @type {boolean}
+     */
+    this.deleted = false;
```

Comment:
Well, discord.js constructing a deleted guild scheduled event doesn't show it's deleted here because this boolean is never actually modified. Interesting!

Reply:
It gets modified in the action handler:
https://github.com/discordjs/discord.js/blob/611fd63845d8901f6a0db2bf789088057463e3f0/src/client/actions/GuildScheduledEventDelete.js#L15

Reply:
Not when fetched from the audit logs.

Reply:
fixed